src/plopp/core/limits.py

```diff
@@ -1,51 +1,86 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
-from typing import Literal, Tuple
+from typing import Literal, Optional, Tuple
 
 import numpy as np
-from scipp import Variable, scalar
+import scipp as sc
+
+from .utils import merge_masks
 
 
 def find_limits(
-    x: Variable, scale: Literal['linear', 'log'] = 'linear'
-) -> Tuple[Variable, ...]:
+    x: sc.DataArray,
+    scale: Optional[Literal['linear', 'log']] = 'linear',
+    pad: bool = False,
+) -> Tuple[sc.Variable, sc.Variable]:
     """
     Find sensible limits, depending on linear or log scale.
     If there are no finite values in the array, raise an error.
     If there are no positive values in the array, and the scale is log, fall back to
     some sensible default values.
     """
+    if scale is None:
```

Comment:
Minor: would it be nicer to leave `scale` as `linear/log` here and let the caller worry about the `None` case?